src/actions/assetsActions.js

```diff
@@ -383,112 +376,62 @@ export const fetchAllAccountsTotalBalancesAction = () => {
     const accounts = accountsSelector(getState());
     const smartWalletAccounts = accounts.filter(isNotKeyBasedType);
 
-    smartWalletAccounts.forEach((account) => dispatch(fetchCollectiblesAction(account)));
+    await Promise.all(smartWalletAccounts.map(async (account) => {
+      dispatch(fetchCollectiblesAction(account));
 
-    const accountsAddresses = smartWalletAccounts.map((account) => getAccountAddress(account));
-    const availableChainProtocols = await getZapperAvailableChainProtocols(accountsAddresses);
-    if (!availableChainProtocols) {
-      dispatch({ type: SET_FETCHING_TOTAL_BALANCES, payload: false });
-      reportErrorLog('fetchAllAccountsTotalBalancesAction failed: no availableChainProtocols', { accountsAddresses });
-      return;
-    }
+      const accountId = getAccountId(account);
+      const accountAddress = getAccountAddress(account);
 
-    try {
-      await Promise.all(availableChainProtocols.map(async ({ chain, zapperProtocolIds, zapperNetworkId }) => {
-        // we don't need to pull multi chain balances for Archanova account, only Ethereum
-        const requestForAddresses = chain !== CHAIN.ETHEREUM
-          ? accountsAddresses.filter((address) => !isArchanovaAccountAddress(address, accounts))
-          : accountsAddresses;
-
-        const chainProtocolsBalances = await Promise.all(zapperProtocolIds.map(async (zapperProtocolId) => {
-          const protocolBalancesByAccounts = await getZapperProtocolBalanceOnNetwork(
-            requestForAddresses,
-            zapperProtocolId,
-            zapperNetworkId,
-          );
-
-          const balanceCategory = mapZapperProtocolIdToBalanceCategory(zapperProtocolId);
-          if (!protocolBalancesByAccounts || !balanceCategory) return null;
-
-          return { category: balanceCategory, balances: protocolBalancesByAccounts };
+      // we're fetching and storing values in USD and converting rates by app selected currency later
+      const accountTotalBalances = await etherspotService.getAccountTotalBalances(accountAddress, USD);
+      if (!accountTotalBalances) return;
+
+      accountTotalBalances.forEach(({
+        balances,
+        category: balancesCategory,
+        chainId,
+        totalBalance,
+      }) => {
+        const chain = chainFromChainId[chainId];
+        const assetsCategory = assetsCategoryFromEtherspotBalancesCategory[balancesCategory];
```

Comment:
Pls check for unexpected balanacesCategory that would return undefined assetsCategory.

Reply:
There won't be as it's directly mapped, please check Etherspot `AccountDashboardProtocols` schema (https://docs.etherspot.dev/enums/accountdashboardprotocols.html).

Will still add a check tho just in case.

Reply:
We're not mapping rewards, so if backend update causes it to return rewards we will get `undefined` here. Same if for some reason they will add any other category in the future, which could affect existing clients.